lighthouse-cli/test/smokehouse/report-assert.js

```diff
@@ -151,14 +151,50 @@ function makeComparison(name, actualResult, expectedResult) {
   };
 }
 
+/**
+ * Delete expectations that don't match environment criteria.
+ * @param {LH.Result} lhr
+ * @param {Smokehouse.ExpectedRunnerResult} expected
+ */
+function filterExpectations(lhr, expected) {
+  const userAgent = lhr.userAgent;
+  const userAgentMatch = /Chrome\/(\d+)/.exec(userAgent); // Chrome/85.0.4174.0
+  if (!userAgentMatch) throw new Error('Could not get chrome version.');
+  const actualChromeVersion = Number(userAgentMatch[1]);
+
+  /**
+   * @param {*} obj
+   */
+  function failsChromeVersionCheck(obj) {
+    if (!obj._minChromeMajorVersion) return false;
+    return actualChromeVersion < obj._minChromeMajorVersion;
+  }
+
+  /**
+   * @param {*} obj
+   */
+  function filter(obj) {
+    for (const key of Object.keys(obj)) {
+      if (!obj[key] || typeof obj !== 'object') continue;
+      else if (failsChromeVersionCheck(obj[key])) delete obj[key];
+      else filter(obj[key]);
+    }
+    delete obj._minChromeMajorVersion;
+  }
+
+  filter(expected);
```

Comment:
🚲 🏠 filter tends to make me think it returns a new thing, when we delete stuff elsewhere we call it `prune` maybe?

not a big deal

then fn name of `pruneNewerChromeExpectations` or something :D